hummingbot/connector/derivative/bybit_perpetual/bybit_perpetual_derivative.py

```diff
@@ -451,10 +451,13 @@ async def _all_trade_updates_for_order(self, order: InFlightOrder) -> List[Trade
         if order.exchange_order_id is not None:
             try:
                 all_fills_response = await self._request_order_fills(order=order)
-                fills_data = all_fills_response["result"]["trade_list"]
-                for fill_data in fills_data:
-                    trade_update = self._parse_trade_update(trade_msg=fill_data, tracked_order=order)
-                    trade_updates.append(trade_update)
+                trades_list_key = "data" if bybit_utils.is_linear_perpetual(order.trading_pair) else "trade_list"
+                fills_data = all_fills_response["result"].get(trades_list_key, [])
+
+                if fills_data is not None:
```

Comment:
I think that this condition won't be reached because if you don't find the key, will return an empty list not a None value.

Reply:
I think this is to safe guard against the eventuality that `all_fills_response = {"result": None}`, in which case `fills_data` will be `None`